PowerEditor/src/WinControls/ProjectPanel/TreeView.cpp

```diff
@@ -364,6 +388,10 @@ void TreeView::moveTreeViewItem(HTREEITEM draggedItem, HTREEITEM targetItem)
 	tvInsertStruct.hParent = targetItem;
 
 	HTREEITEM hTreeParent = (HTREEITEM)::SendMessage(_hSelf, TVM_INSERTITEM, 0, (LPARAM)(LPTVINSERTSTRUCT)&tvInsertStruct);
+	_validHandles.insert(hTreeParent);

```

Comment:
You ***absolutely*** should check to see if `::SendMessage` returned `NULL` before using `newItem`.

***Especially*** since you're putting it in a variable named `_validHandles`.